gitlab/base.py

```diff
@@ -132,7 +132,7 @@ def __ne__(self, other: object) -> bool:
         return super(RESTObject, self) != other
 
     def __dir__(self):
-        return super(RESTObject, self).__dir__() + list(self.attributes)
+        return super(RESTObject, self).__dir__() | self.attributes.keys()
```

Comment:
Can the result be converted to a list? As that is what it returned before.

```
    return list(super(RESTObject, self).__dir__() | self.attributes.keys())
```

Reply:
The `__dir__` method can return any sequence and will be converted into a list and sorted automatically. For example, if you return a set or any other sequence, `dir(obj)` will always be a list.

```python
class MyObj:
    def __dir__(self):
        return {'abc', 'xyz', 'def',}
obj = MyObj()
dir(obj)
['abc', 'def', 'xyz']
```

If you like we can convert it in the code, if for nothing else, to be more explicit.

Reply:
I (for selfish reasons) would like it to return a list. As I am going to be working on adding type-hints to that code in the future 😊 But not necessary.

Reply:
And not necessary, because I think I will be adding a type-hint for the return value of `Iterable[something]`, most likely.

Reply:
Sounds good to me. Feel free to apply the above suggestion (or not)

Looking forward to type hinting!

Worth mentioning that it's _usually_ not necessary to provide type annotations or type hints for certain magic methods, since their return type is implied because there's no way to implement them to return a different type.

For example, the return type for methods like `__str__`, `__repr__`, `__len__`, `__dir__`, are known to IDEs and type checkers like MyPy automatically because the returned type must be valid, otherwise a `TypeError` is raised.